backend/controllers/auth_controller.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ZPI-2024-25/KubernetesAccessManager/auth"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+func CheckLoginStatus(w http.ResponseWriter, r *http.Request)  {
+	token, err := auth.GetJWTTokenFromHeader(r)
+	isValid, claims := auth.IsTokenValid(token)
+
+	if err != nil || !isValid {
+		writeJSONResponse(w, http.StatusUnauthorized, models.ModelError{
+			Message: "Unauthorized",
+			Code: http.StatusUnauthorized,
+		})
+		return
+	}
+	rolemap, err := auth.GetRoleMapInstance()
+	if err != nil {
+		writeJSONResponse(w, http.StatusInternalServerError, &models.ModelError{
+			Message: "Internal Server Error",
+			Code: http.StatusInternalServerError,
+		})
+	}
+	status, errM := getLoginStatus(claims, rolemap)
+	if errM != nil {
+		writeJSONResponse(w, int(errM.Code), errM)
+		return
+	}
+	writeJSONResponse(w, http.StatusOK, status)
+}
+
+func getLoginStatus(claims *jwt.MapClaims, rolemap *auth.RoleMapRepository) (*models.UserStatus, *models.ModelError) {
+	roles, err := auth.ExtractRoles(claims)
+	if err != nil {
+		return nil, &models.ModelError{
+			Message: "Roles not found in bearer token",
+			Code: http.StatusInternalServerError,
+		}
+	}
+	access := rolemap.GetAllPermissions(roles)
+	auth.PrunePermissions(access)
+	permissions := shortenPermissions(access)
+	exp, preferredUsername, email := auth.ExtractUserStatus(claims)
+
+	return &models.UserStatus{
+		Permissions: permissions,
+		User: &models.UserStatusUser{
+			PreferredUsername: preferredUsername,
+			Email: email,
+			Exp: exp,
+		},
+	}, nil
+}
+
+func shortenPermissions(pmatrix map[string]map[string]map[models.OperationType]struct{}) map[string]map[string][]string {
+	res := make(map[string]map[string][]string)
+
+	for k, v := range pmatrix {
+		res[k] = make(map[string][]string)
+		for k2, v2 := range v {
+			res[k][k2] = make([]string, len(v2))
+			i := 0
+			for k3 := range v2 {
+				res[k][k2][i] = shorterP(k3)
+				i++
+			}
+		}
+	}
+	return res
+}
+
+func shorterP(p models.OperationType) string {
+	switch p {
+		case models.Create:
+			return "c"
+		case models.Read:
+			return "r"
+		case models.Update:
+			return "u"
+		case models.Delete:
+			return "d"
+		case models.List:
+			return "l"
+		default:
+			return "x"
+		}
```

Comment:
These 2 functions needs to be changed - neither of them, in my opinion, conveys by name what it really does - one changes the type of structure from a map of keys, to a map of allowed operations, and the other maps operations to the corresponding letters. shortenPermissions is written in such a way that without context it is not clear what it does, and even with it it is difficult to tell what happens when. Why can't k and v be renamed to names that tell you what keys and what values are being processed? Another thing is that, at least in my opinion, the authorization controller is not the right place for them. This are functions that are inextricably linked to operations and could be used elsewhere. I don't understand why it wasn't added to type_operation.go, since there are already other functions there, but another thing is that I'm not a fan of making any functions in models. In summary, it would be useful to split functions about auth into more files and use more verbose variable names.

Reply:
And more verbose function names.

Reply:
I've cleaned up the shorterP mess hope u like it  ▄︻デ══━一